store/watcher_hub.go

```diff
@@ -61,7 +63,9 @@ func (wh *watcherHub) watch(key string, recursive, stream bool, index, storeInde
 	}
 
 	w := &watcher{
-		eventChan:  make(chan *Event, 100), // use a buffered channel
+		// The eventChan needs to be at least as big as the event history to support streaming
+		// events from an outdated index.
+		eventChan:  make(chan *Event, wh.EventHistory.Queue.Capacity),
```

Comment:
It's populated with one event history scan when we add the watcher, but it empties out when we send the events across the stream watch. after that each notify will only put one event in there right? For non-stream watchers the scan should only return a single event. 

Of course the space is still used. Do you think we should instead return 100 events and let clients re-open the watch with a new index?